src/cpphttplib_http_async_writer_impl.cpp

```diff
@@ -0,0 +1,94 @@
+//*****************************************************************************
+// Copyright 2024 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "cpphttplib_http_async_writer_impl.hpp"
+
+#include <functional>
+#include <string>
+#include <utility>
+
+#include "logging.hpp"
+
+namespace ovms {
+
+// Used by V3 handler
+void CppHttpLibHttpAsyncWriterImpl::OverwriteResponseHeader(const std::string& key, const std::string& value) {
+    SPDLOG_DEBUG("CppHttpLibHttpAsyncWriterImpl::OverwriteResponseHeader {} {}", key, value);
+    // TODO
+}
+void CppHttpLibHttpAsyncWriterImpl::PartialReplyWithStatus(std::string message, HTTPStatusCode status) {
+    SPDLOG_DEBUG("CppHttpLibHttpAsyncWriterImpl::PartialReplyWithStatus {} {}", message, int(status));
+    // TODO
+}
+void CppHttpLibHttpAsyncWriterImpl::PartialReplyBegin(std::function<void()> cb) {
```

Comment:
Q: Shouldn't this cb be rvalue?

Reply:
cpp-httplib perf was not acceptable, therefore this will be deleted from the PR anyway

however, the lambda is std::moved by the caller anyway